### PR TITLE
C# reference page

### PR DESCRIPTION
**Pull request by alex-tpom6oh**

### CODE REVIEW

File: pages/implementation/reference/java.mdx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Link from '../../../components/Link';
 
 Avo can code generate Avo Functions in Java targeted at the following platforms
 
-- Android
+- Androider parameters de
```
Comment (alex-tpom6oh) [inline pages/implementation/reference/java.mdx:12]:
```suggestion
- Android
```

**Comment by hugihlynsson** [comment]
#### 📝Changed routes:


| [**…ation/reference/csharp**](https://docs-5dyahggci.avo-dev.app/implementation/reference/csharp) | [**…ntation/reference/java**](https://docs-5dyahggci.avo-dev.app/implementation/reference/java) |
|:-:|:-:|
|<a href="https://docs-5dyahggci.avo-dev.app/implementation/reference/csharp">Screenshot of /implementation/reference/csharp</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/qR_mm0nCjs-uXNJVLRj47rOhfJSvWXFjkoUZjsP_CvvaldfGqNIj2x6U0gt4DWbz2NSU2gzztgV47Jla8brruw.png">(view full size)</a> |<a href="https://docs-5dyahggci.avo-dev.app/implementation/reference/java">Screenshot of /implementation/reference/java</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/WPNnf5GT-VCdaFmYIo0RBXVST6B-64_eBUpuKj7FrTWbTZIFWyyBbhWPKOfcDYIeeHG_fTmdAln6aa4z8ObdSQ.png">(view full size)</a> |



| [**…-programming-languages**](https://docs-5dyahggci.avo-dev.app/implementation/supported-programming-languages) |
|:-:|
|<a href="https://docs-5dyahggci.avo-dev.app/implementation/supported-programming-languages">Screenshot of /implementation/supported-programming-languages</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/T23ps6cdlgYcvP8xlrtEMxm7ejYL3qWJ_46-W2H7PzrpjMpU4CJJ5jUh5Wg_YvcrHb2eYnmFHtNNQXZGd_Z1dw.png">(view full size)</a> |







Commit edb4e0b3ac2655590340f5c975c75316c54ac826 (https://docs-5dyahggci.avo-dev.app).

File: pages/implementation/reference/csharp.mdx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+You can also <Link passHref href="/explore-tracking-plan/download-or-copy-avo-file-manually"><a>download the file manually</a></Link> from your Avo workspace.
+
+#### Step 2. Initialize Avo
+
+Initialize Avo by creating an object using constructor from the generated Avo file
+
+```csharp
+using Avo;
+
+Avo.Avo avo = new Avo.Avo(Avo.AvoEnv.Dev/*, other parameters depending on your tracking plan setup*/);
```
Comment (bragur) [inline pages/implementation/reference/csharp.mdx:38]: AvoAvo.Avo.avo? 😂

Comment (alex-tpom6oh) [reply pages/implementation/reference/csharp.mdx:38]: AVO!!!111

File: pages/implementation/reference/csharp.mdx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+`strict: Boolean`: bool, if true, Avo will throw an exception when it detects a tracking problem in development or staging. Note that the strict flag is ignored in production.
+
+#### setAvoLogger
+
+```csharp
+public static ILogger __LOGGER__ = null;
+```
+
+This filed allows you to provide custom implementation of the logger used by the Avo Functions.
```
Comment (bragur) [inline pages/implementation/reference/csharp.mdx:124]:
```suggestion
This field allows you to provide a custom implementation of the logger used by the Avo Functions.
```

File: pages/implementation/reference/csharp.mdx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+using Avo;
+
+Avo.Avo avo = new Avo.Avo(Avo.AvoEnv.Dev/*, other parameters depending on your tracking plan setup*/);
+```
+
+The actual parameters depend on your tracking plan setup, see the parameters explanation in <Link passHref href="/implementation/reference/csharp#constructor"><a>the reference below</a></Link>.
+
+#### Step 3. Call Avo functions to track your product usage
+
+Every event in your tracking plan, marked with the "Implement with Avo" checkbox, gets a function in the generated code, named according to the event name, in PasclaCase.
```
Comment (bragur) [inline pages/implementation/reference/csharp.mdx:45]:
```suggestion
Every event in your tracking plan, marked with the "Implement with Avo" checkbox, gets a function in the generated code, named according to the event name, in PascalCase.
```

File: pages/implementation/reference/csharp.mdx [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+```csharp
+public void [YourEventName](int eventProperty0, double? eventProperty1, ..., bool? userProperty0, IList<int> userProperty1, ..., string userId_, string anonymousId_)
+```
+
+Every event you define in your tracking plan in Avo gets a function named after the event in PascalCase. The arguments of the function depend on how it's defined in your tracking plan
+
+##### Arguments
+
+`eventProperty: type defined in the Avo tracking plan, can be string, int, long, float, bool, list, object and any`:
```
Comment (bragur) [inline pages/implementation/reference/csharp.mdx:154]: Is this large code block intentional?

**Review by bragur — APPROVED** [review]
Just a few comments

**Review by bragur — APPROVED** [review]

File: pages/implementation/reference/csharp.mdx (hunk 5 above)
Comment (alex-tpom6oh) [reply pages/implementation/reference/csharp.mdx]: Yes, not the best idea probably :)

**Pull request merged**